### PR TITLE
Fix regression with trello api get requests.

### PR DESCRIPTION
**Pull request by GMartin-dev**
Fix for the issue:
https://github.com/sarumont/py-trello/issues/373

### CODE REVIEW

File: trello/trelloclient.py [hunk 1]
```diff
@@ -212,13 +212,14 @@ def fetch_json(
             headers = {}
         if query_params is None:
             query_params = {}
-        if post_args is None:
-            post_args = {}
 
-        # if files specified, we don't want any data
+        # Per trello api specificacion payload should not be present on get requests.
```
Comment (BonySmoke) [inline trello/trelloclient.py:216]: small typo:)
specificacion -> specification

**Comment by Edren06** [comment]
Thanks!

**Comment by beretem** [comment]
When can we expect to be merged and a new pip version published?

File: trello/trelloclient.py [hunk 2]
```diff
@@ -212,13 +212,14 @@ def fetch_json(
...
         data = None
+        
+        # if files specified, we don't want any data
         if files is None:
-            data = json.dumps(post_args)
+            if post_args is not None:
```
Comment (chaami) [inline trello/trelloclient.py:221]: I would rather check if it's a get so you don't send body data on a get even if some has been specified, which could be achieved with the line below instead:
```
if post_args and http_method != 'GET'
```

**Comment by chaami** [comment]
Hi @sarumont,
The deployment of this [Trello's change](https://developer.atlassian.com/changelog/#CHANGE-1459) makes it essentially impossible to use Trello with the package.
Could you please have a look at this pull request and its associated [issue](https://github.com/sarumont/py-trello/issues/373) ?

Thanks in advance

**Comment by aboudaty** [comment]
Hi @sarumont,

Sorry to chase, have you had a chance to look at this, please? It's not possible to use the package without a fix now.

Many thanks

**Comment by Kab00mKap0w** [comment]
See [discussion](https://github.com/sarumont/py-trello/issues/376#issue-2218799609) re: repo's future

File: trello/trelloclient.py (hunk 2 above)
Comment (rudjDfi) [reply trello/trelloclient.py:221]: I gree

**Comment by Konano** [comment]
For personal needs, I have forked this repository and merged this pending PR into the new repository.
https://github.com/Konano/py-trello-api

As a temporary solution, I have also uploaded the new package to Pypi. You can download the fixed py-trello with this command:
`pip install py-trello-api==0.20.0`

I hope this helps someone in need.

File: trello/trelloclient.py (hunk 2 above)
Comment (ogarcia) [reply trello/trelloclient.py:221]: Better:

```python
data = None if files is not None or method == 'GET' else json.dumps(post_args)
```

Data must have None if:
- Files has value (is not None)
- method is GET regardless data value (cannot post data or files using a GET)

With this method it is not necessary to remove the initial section of code:
```python
if post_args is None:
  post_args = {}
```

**Comment by spectromas** [comment]
Registering my eager interest in this being merged

**Pull request merged**